app.js

```diff
@@ -2,14 +2,45 @@ const express = require("express");
 const app = express();
 app.use(express.json());
 const { getTopics } = require("./controllers/get_topics.controller");
+const {
+  getArticlesById,
+} = require("./controllers/get-articles-by-id.controller");
 
 app.get("/api/topics", getTopics);
 
+app.get("/api/articles/:article_id", getArticlesById);
+
+
 app.use("/", (req, res) => {
   res.status(404).send({ msg: "not found" });
 });
 
+
+app.use((err, req, res, next) => {
+  if (err.status === 404 && err.msg) {
+    res.status(404).send(err);
+  }
+
+  next(err);
+});
+
+app.use((err, req, res, next) => {
+  if (err.status === 400 && err.msg) {
+    res.status(400).send(err)
+  }
+
+  next(err);
+});
+
+app.use((err, req, res, next)=>{
+  if(err.code === "22P02"){
+  res.status(400).send({msg: "invalid id"})
+      }
+  next(err)
+      })
+
```

Comment:
I would move those error handling middleware to a separate file so everything looks more organised